dwave/gate/simulator/simulator.pyx

```diff
@@ -145,30 +153,106 @@ def simulate(
     state[0] = 1
 
     for op in circuit.circuit:
-        targets = [circuit.qubits.index(qb) for qb in op.qubits]
-        apply_instruction(num_qubits, state, op, targets, little_endian)
+        if isinstance(op, ops.Measurement):
+            _measure(op, state, circuit)
```

Comment:
Why treat `_measure` differently here? I think it should go with the rest of the instructions in `apply_instruction`.

Reply:
Fair enough. I think the reason I put it here was to avoid having to loop over the target qubits a second time for measurments (in the `_measure` function), but it's a very small overhead.